nodes/wot-discovery.js

```diff
@@ -147,6 +154,74 @@ module.exports = function (RED) {
 
             return addresses;
         }
+        function _getDiscoveryLinks(){
+            if (config.useCoap) {
+                coapAddresses.forEach(address => {
+                    const reqOpts = url.parse(`coap://${address}/.well-known/core`);
+                    reqOpts.pathname = reqOpts.path;
+                    reqOpts.query ="rt=wot.thing";
+                    reqOpts.multicast = true;
+                    const req = coap.request(reqOpts);
+                    req.on("response", _coreResponse);
+                    req.on("error", function (err) {
+                        node.log("client error");
+                        node.log(err);
+                    });
+                    req.end();
+                });
+            }
+        }
+
+        function _coreResponse(res){
+            res.on("data", (data) => {
+                 if (res.headers["Content-Format"] === "application/link-format") {
+                    let links = data.toString().split(',');
+                    
+                    links = links.map(link => {
+                        return link.split(";");
+                    });
+                    
+                    links.forEach(link => {
+
+                        let correctResourceType = false;
+                        let correctContentType = false;
+                        let path;
+
+                        link.forEach(function (curentValue, index) {
+                            if (index === 0) {
+                                // First parameter must be the path
+                                // TODO: Add assertion for </ ... > format
+                                path = curentValue.substring(2, curentValue.length - 1);
+                                return;
+                            } else if (!path) {
+                                return;
+                            }
+
+                            curentValue = curentValue.split("=");
+                            let parameter = curentValue[0];
+                            let values = curentValue[1]
+
+                            switch (parameter) {
+                                case "ct":
+                                    if (values === "432") {
+                                        correctContentType = true;
+                                    }
+                                    break;
+                                case "rt":
+                                    if (values === '"wot.thing"') {
+                                        correctResourceType = true;
+                                    }
+                                    break;
+                            }
+                        })
+
+                        if (correctContentType && correctResourceType) {
+                            _sendCoapDiscovery(`[${res.rsinfo.address}]`, path);
```

Comment:
One minor thing: In the [RFC](https://datatracker.ietf.org/doc/html/rfc6690#section-2) I noticed that the URI-Reference (here currently denoted as "path") can also be an absolute one. Therefore I think that we have to perform some parsing (probably using `urllib`) to determine if it is a relative or an absolute URI.

Reply:
I added a ToDo in the description for this.